src/open_inwoner/qmatic/client.py

```diff
@@ -0,0 +1,141 @@
+    duration: int
+    additionalCustomerDuration: int
+    custom: str | None
+
+
+class FullServiceDict(ServiceDict):
+    active: bool
+    publicEnabled: bool
+    created: int
+    updated: int
+
+
+class ServiceGroupDict(BaseModel):
+    services: list[ServiceDict]
+
+
+class BranchDict(BaseModel):
+    branchPublicId: str
+    branchName: str
+    serviceGroups: list[ServiceGroupDict]
+
+
+class BranchDetailDict(BaseModel):
+    name: str
+    publicId: str
+    phone: str | None
+    email: str | None
+    branchPrefix: str | None
+
+    addressLine1: str | None
+    addressLine2: str | None
+    addressZip: str | None
+    addressCity: str | None
+    addressState: str | None
+    addressCountry: str | None
+
+    latitude: float | None
+    longitude: float | None
+    timeZone: str
+    fullTimeZone: str
+    custom: str | None
+    created: int
+    updated: int
+
+
+class Appointment(JSONEncoderMixin, BaseModel):
+    services: list[ServiceDict]
+    title: str
+    start: datetime
+    end: datetime
+    created: int
+    updated: int
+    publicId: str
+    branch: BranchDetailDict
+    notes: str | None
+
+    class Config(JSONEncoderMixin.Config):
+        pass
+
+
+class NoServiceConfigured(RuntimeError):
+    pass
+
+
+# API CLIENT IMPLEMENTATIONS, per major version of the API
+
+
+def QmaticClient() -> "Client":
+    """
+    Create a Qmatic client instance from the database configuration.
+    """
+    config = QmaticConfig.get_solo()
+    assert isinstance(config, QmaticConfig)
+    if (service := config.service) is None:
+        raise NoServiceConfigured("No Qmatic service defined, aborting!")
+    assert isinstance(service, Service)
+    return build_client(service, client_factory=Client)
+
+
+def startswith_version(url: str) -> bool:
+    if url.startswith("v1/"):
+        return True
+    if url.startswith("v2/"):
+        return True
+    return False
+
+
+class Client(APIClient):
+    """
+    Client implementation for Qmatic.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.headers["Content-Type"] = "application/json"
+
+    def request(self, method: str, url: str, *args, **kwargs):
+        # ensure there is a version identifier in the URL
+        if not startswith_version(url):
+            url = f"v1/{url}"
+
+        response = super().request(method, url, *args, **kwargs)
+
+        if response.status_code == 500:
+            error_msg = response.headers.get(
+                "error_message", response.content.decode("utf-8")
+            )
+            raise QmaticException(
+                f"Server error (HTTP {response.status_code}): {error_msg}"
+            )
+
+        return response
+
+    def list_appointments_for_customer(
+        self, customer_externalid: str
+    ) -> list[Appointment]:
+        endpoint = f"customers/externalId/{customer_externalid}/appointments"
```

Comment:
Are we confident we can build URL's with just f-string instead of something that applies URL-encoding?